src/mots/config.py

```diff
@@ -66,18 +71,74 @@ def load(self):
         with self.path.open("r") as f:
             self.config = yaml.load(f)
 
-    def write(self):
-        """Write configuration to file, and update the timestamp."""
+    def check_hashes(self) -> List[str]:
+        """Check that the hashes in the config are up to date.
+
+        Upon calling this function, the existing configuration is copied and stripped
+        of volatile keys, then a hash is calculated and compared against the old hash.
+
+        If there is a mismatch, return non-zero exit code. Otherwise return 0.
+        """
+        errors = []
+        config = self.config.copy()
+
+        # Exclude hashes and updated timestamp from hash generation
+        hashes = config.pop("hashes", {})
+        config.pop("updated_at", None)
+
+        # Write actual config yaml dump to stream.
+        with io.StringIO() as stream:
+            yaml.dump(config, stream)
+            content = stream.getvalue()
```

Comment:
Info: have you checked that the yaml library preserves order when loading/writing the files? If it doesn't this could result in different calculated hashes despite the same content.

This solution will ignore comment changes (since those won't be loaded into `self.config`) but would require an update if someone moved an entry in the file but kept the content the same. Is that okay/expected?

Reply:
Order is preserved as this is one of the reasons `ruamel.yaml` was chosen (has good round-trip support in various use cases). There are also various things that are resorted, resulting in a more consistent output (`people` and `modules` for example).

Comment changes also have no impact on any functionality, thus safe to be ignored as well. In the case of moving things around that have comments in them, there are some cases where this could result in a change but in many cases (e.g. moving a module) it will be resorted anyway. In fact, the order of keys in each module definition should also be fixed, which I will implement in a separate bug.